stdin-wrapper

```diff
@@ -6,15 +6,15 @@
 #   temporary file, and the first argument sets a particular file extension
 #   for the temporary file.
 
+set -x
+
 # All of the following arguments are read as command to run.
 file_extension="$1"
 shift
 
 temp_file=$(mktemp --tmpdir "ale-XXXXXXXXX$file_extension")
 trap 'rm $temp_file' EXIT
 
-while read -r; do
-    echo "$REPLY" >> "$temp_file"
-done
+cat - > "$temp_file"
```

Comment:
Instead of creating temporary file and writing data to it provide named pipe for the process to "fool it" that it is reading file, not stream. This is [reaison d'etre of named pipes](https://groups.google.com/d/msg/golang-dev/R9hbrKG18hQ/6RjEiOOG_hoJ).